src/main/java/io/confluent/kafkarest/Errors.java

```diff
@@ -51,6 +51,13 @@ public static RestException consumerInstanceNotFoundException() {
                                      CONSUMER_INSTANCE_NOT_FOUND_ERROR_CODE);
   }
 
+  public final static String BROKER_DATA_NOT_FOUND_MESSAGE = "Broker data not found.";
```

Comment:
I'm trying to think forward to the new consumer implementation of this same feature and I don't think the errors will match up well. The metadata discovery in the new clients is different from the old ones since they request metadata from the brokers rather than getting it from ZK. It looks like we're correctly detecting when a topic or partition doesn't exist, so I think the only error should be the same in both old and new clients -- the leader is not available. This is represented by LeaderNottAvailableException in the new clients.

So maybe just rename this so we'll be compatible moving forward? I did a quick scan of o.a.k.common.errors and I think that's the only relevant exception, but you might want to also give those classes a once over.